src/Integration/Monolog/CloudWatch/src/CloudWatchLogsHandler.php

```diff
@@ -126,10 +126,20 @@ protected function getDefaultFormatter(): FormatterInterface
     }
 
     /**
-     * {@inheritdoc}
+     * @param \Monolog\LogRecord|array $record
      */
-    protected function write(array $record): void
+    protected function write($record): void
     {
+        if (version_compare(PHP_VERSION, '8', '>=')) {
```

Comment:
Why do we need to check the PHP version?

Reply:
tests fails when i switch down to older version of php needs to handle the cases separately

Reply:
I believe there is something wrong elsewhere.

`if ($arg instanceof Bar) {` works since at least PHP 7.0

https://3v4l.org/Ua4sF#v7.0.33

Reply:
fixed